### ISSUE TITLE
Update to ncurses6

### ISSUE THREAD

**Issue by Yuji-Shimohira-Calvo** [kylebarron/stata_kernel#374]
**Is your feature request related to a problem? Please describe.**
On a fresh (Linux) install Stata Kernel dies due to libtinfo.so.5 be missing in my system.

**Describe the solution you'd like**
ncurses 6 (current version) is backwards compatible with ncurses 5. It'd be nice if libtinfo.so.6 is called instead.

**Describe alternatives you've considered**
A symlink works just fine if libtinfo.so.5 isn't present but libtinfo.so.6 is, but this is rather a lazy approach (?) Of course, installing ncurses 5 is yet another alternative, but since they are backwards compatible...

**Additional context**
Not really sure how many Linux systems pack ncurses 5. In my case (Arch) there is an AUR package but the official package is ncurses 6. I am not sure either if this is a "problem" specific to Stata Kernel. I found it when doing a fresh install and thought it might be worth sharing since a Google search didn't yield much.

**Comment by mcaceresb** [comment]
@Yuji-Shimohira-Calvo I use Arch myself; there's a fairly straightforward fix:

```
ln -sf /usr/lib/libtinfo.so.6 /usr/lib/libtinfo.so.5
```

Similarly for ncurses:

```
ln -sf /usr/lib/libncursesw.so.6 /usr/lib/libncursesw.so.5
ln -sf /usr/lib/libncursesw.so.6 /usr/lib/libncurses.so.5
```

I don't think the Stata kernel can tell Stata itself which libraries to call, however.

**Comment by Yuji-Shimohira-Calvo** [comment]
@mcaceresb Absolutely, as I said above a symlink works perfectly fine, that's how I have it working currently. It's a "subpar" workaround though. But because version 6 of ncurses is backwards compatible with version 5, ideally the latter shouldn't be called. Not a bug per se, more of an improvement suggestion.

**Comment by kylebarron** [comment]
I'm guessing it's Stata's console program itself that requires ncurses. I don't think the kernel does in any way

**Comment by mcaceresb** [comment]
> @mcaceresb Absolutely, as I said above a symlink works perfectly fine, that's how I have it working currently. It's a "subpar" workaround though. But because version 6 of ncurses is backwards compatible with version 5, ideally the latter shouldn't be called. Not a bug per se, more of an improvement suggestion.
> 
> Edit: just refering to a question implicit in OP: who's calling for ncurses5 in this case?

@Yuji-Shimohira-Calvo As Kyle said, the kernel doesn't call it. This is Stata calling ncurses5, and they are the ones who would have to update. Ideally they would call whichever library is installed in the system, given ncurses5 seems to be default in some systems still.
